adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleNativeAd.m

```diff
@@ -43,11 +43,17 @@ @implementation GADMediationVungleNativeAd {
 @synthesize desiredPlacement;
 
 - (void)dealloc {
-  _adConfiguration = nil;
-  _adLoadCompletionHandler = nil;
-  _delegate = nil;
-  _mediaView = nil;
-  _nativeAd = nil;
+    [NSThread isMainThread] ? [self cleanupResources] : dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
In 6.x, there were dispatches in the dealloc and that lead to crashes, are there any of those concerns here?

Reply:
@stanley-vungle yes, I am hesitant to suggest these based on previous experience. I ran some tests and can confirm that the test app did crash when attempting to use weak and strong references in dealloc. The current approach (directly referencing self with the block) seems to work with no issue in test. I ran the code through GPT to validate - in the response, statement is that even if the main thread execution is delayed and self is nil by the time the block executes, it should not trigger a crash because it would only be attempting to send a message to nil.

All that being said, if there's concern about shipping this alongside other changes in 7.4.1, we can hold on this and submit this after the official release PR.

CC @ManojBudumuru-vungle ^^ - any thoughts?

Reply:
I think it should be fixed in the SDK instead of dispatching to another thread in the dealloc. But if you think this solution is stable, then I'll take your word for it

Reply:
i am fine with this. but if we can fix it in sdk we should probably do that for 7.5 ?